addon/content/reducer/reducer-summary.js

```diff
@@ -49,7 +50,10 @@ async function handleShowDetails(messages, state, dispatch, updateFn) {
 export const summaryActions = {
   replaceConversation({ summary, messages }) {
     return async (dispatch, getState) => {
-      await handleShowDetails(messages, getState(), dispatch, () => {
+      await handleShowDetails(messages, getState(), dispatch, async () => {
+        // The messages inside `msgData` don't come with filled in `to`/`from`/ect. fields.
+        // We need to fill them in ourselves.
+        await mergeContactDetails(messages.msgData);
```

Comment:
Btw, what options do we have for testing the reducers or maybe the display of contacts? I'd like to get some tests set up for the contacts display fairly soon, as I think this is quite complicated, and we should make sure we can get/keep it right.